Automattic/jetpack
# Sharing / Likes / Related Posts: do not display markup in ActivityPub requests

Fixes #29435

## Proposed changes:

See the details in the issue for more information behind that change.

This is fixed using a recent change in the ActivityPub plugin:
https://github.com/Automattic/wordpress-activitypub/pull/328

**Before**

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/426388/224988365-4eb1b53b-31e8-4bfc-9961-abb5375bc623.png">

**After**

<img width="1223" alt="image" src="https://user-images.githubusercontent.com/426388/224988158-47dd6ed9-2052-4cee-bcda-50867bab7df1.png">

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

> **Note**
> To test, you'll need to be able to make requests to your site. The easiest way to do that is probably via your terminal.

* Start with a site connected to WordPress.com.
* Go to Jetpack > Settings, and activate sharing, likes, and related posts.
* Install and activate the development version of the ActivityPub plugin, available on https://github.com/Automattic/wordpress-activitypub/
* Open a post on the frontend, ensure the sharing buttons as well as the likes are displayed. Related Posts will be displayed too, but only if you have enough published posts on your site for that. Otherwise, you'll only see the Related Posts container div.
* In your terminal, query `curl 'https://yoursite.com/yourpost/' -H 'Accept: application/activity+json'
* Check the output; it should not include the markup from the elements above.
